### ISSUE TITLE
i18n: ICU plurals using "1"

### ISSUE THREAD

**Issue by exterkamp** [GoogleChrome/lighthouse#10786]
The titles for the new diagnostics (although I predict any ICU plural we use with an explicit "1") are not rendered properly with non-standard numeric representations.
![image](https://user-images.githubusercontent.com/6392995/81980798-1e339300-95e4-11ea-97e9-9a02ca90294d.png)

This is because the ICU decleration for the "1" case is explicitly an `en` "1".
```javascript
  displayValue: `{nodeCount, plural,
    =0 {No elements found}
    =1 {1 element found} // this uses the literal "1"
    other {# elements found} // this uses the number, like the character "١"
    }`,
```

**Comment by patrickhulce** [comment]
I'm not sure I understand this one @exterkamp, could you explain a tad more? Isn't it the role of the translators to author the appropriate translated plurals and we're only writing the english ones? 

Based on my understanding, `1 element found` would never use a number other than 1...

**Comment by exterkamp** [comment]
`ar` uses the eastern arabic numeral set: Eastern Arabic  ٠ | ١ | ٢ | ٣ | ٤ | ٥ | ٦ | ٧ | ٨ | ٩ | ١٠
Notice this is "4 elements found", but uses a "٤" instead of "4": 
![image](https://user-images.githubusercontent.com/6392995/82010437-f1ed3600-9626-11ea-8017-d01c7a1246d1.png)
Now look at how it handles "1 element found":
![image](https://user-images.githubusercontent.com/6392995/82010677-9ec7b300-9627-11ea-9265-c6afd4404fbf.png)

It should say "١ element found"; however the ICU plural for `=1` is the literal "1 element found". And I don't think the translators will use eastern arabic numerals consistently with what we do (we always use them). Additionally I doubt they will use thai or traditional Chinese numerics as well if we ever choose to enforce those characters: https://github.com/GoogleChrome/lighthouse/tree/i18n-numbers
![image](https://user-images.githubusercontent.com/6392995/82011863-14814e00-962b-11ea-89a3-3fbce28cb0be.png)
![image](https://user-images.githubusercontent.com/6392995/82011820-f87dac80-962a-11ea-988c-5fd2c2546458.png)
![image](https://user-images.githubusercontent.com/6392995/82011847-07fcf580-962b-11ea-99e3-1300f5ad1520.png)

I think we will need to use the replacement for `=1` as well. Like this:

```javascript
  displayValue: `{nodeCount, plural,
    =0 {No elements found}
    =1 {# element found}
    other {# elements found}
    }`,
```

That said, I don't think this is high priority, since our numeric i18n story is all over the place...
![image](https://user-images.githubusercontent.com/6392995/82012104-c751ac00-962b-11ea-824f-4ac854af502f.png)

**Comment by patrickhulce** [comment]
> Notice this is "4 elements found", but uses a "٤" instead of "4":
> Now look at how it handles "1 element found":

I got that part, what I was asking is why does it matter? For `ar` we won't be using the message `1 element found` we'll be using something else written by the translator that's written in that locale and *that* message should use the correct numeral or `#` in place of it.

> I don't think the translators will use eastern arabic numerals consistently

I think this answer my real question. I am understanding this correctly that the concern is translators will use whatever numeral we write in the english string and not use whatever numeral is appropriate for the locale they are translating to?

**Comment by exterkamp** [comment]
>translators will use whatever numeral we write in the english string and not use whatever numeral is appropriate for the locale they are translating to and/or appropriating substitute # for it

Correct. In the same way that our markdown was getting ruined by some translators replacing " ` " with " ' ". Translating is not as consistent as we want it to be. I think if we try to become more consistent with numerals of other languages, then we should probably look into this as an issue. 

Is it an issue now? _meh_
Is it something that we should handle in the future? _probably_

**Comment by connorjclark** [comment]
can we add a test asserting there is no:

* `
* [
* ]
* (
* )

**Comment by exterkamp** [comment]
> can we add a test asserting there is no:
> 
> * `
> * [
> * ]
> * (
> * )
> 
> maybe not an assert, since we couldn't easily fix it (unless you can just edit it in TC?) maybe just autofix and warn

This seems unrelated. Is there a problem with markdown now? I was talking prior to the ICU escaping & CTC files.

**Comment by connorjclark** [comment]
this is what we have today

<img width="881" alt="image" src="https://user-images.githubusercontent.com/4071474/161842499-c880a4e8-0a46-49a1-997f-86d62513d70e.png">


in general we need to fix numeral i18n

**Comment by connorjclark** [comment]
ok well the numerics are translated properly for other languages in the metrics table.

<img width="876" alt="image" src="https://user-images.githubusercontent.com/4071474/161867954-aca9c2e4-b525-436b-8834-1fc6584a59e4.png">

arabic isn't getting localized numbers because `intl-messageformat` doesn't support it i guess.

We are on an old version of that package, perhaps upgrading is in order.

anyway if we want arabic numerals we should use a different locale code. `ar-u-nu-arab` see [this](https://github.com/formatjs/formatjs/issues/1193) 


not here tho
<img width="357" alt="image" src="https://user-images.githubusercontent.com/4071474/161868023-e4b2bf33-8711-41c0-9108-98826110d31b.png">

**Comment by connorjclark** [comment]
can get arabic numerals from CLI with no code changes:


```
yarn start https://www.example.com/ -A --view --locale=ar-u-nu-arab
```

<img width="855" alt="image" src="https://user-images.githubusercontent.com/4071474/161869186-399771bf-934c-45a9-8570-6e57eaea0fb4.png">
(but not the gauge..)

**Comment by brendankenny** [comment]
> arabic isn't getting localized numbers because intl-messageformat doesn't support it i guess.

FWIW, this comes from the browser's formatting which `intl-messageformat` uses. e.g. `(50).toLocaleString('ar')` gives `'50'` while `(50).toLocaleString('ar-EG')` gives '٥٠'.

Reading about this today, I believe this may be intentional. Basically there are multiple types of digits depending on the region (ar-eg, ar-tn, etc), and the decision was made that `ar` (aka `ar-001` or world-wide `ar`) would use ascii/latin digits as it would be mostly widely understood, even if it was likely not the user's usual choice of digit.

This [CLDR thread](https://unicode-org.atlassian.net/browse/CLDR-10825) gives some background, though I haven't found the follow up yet that moved it from testing to default.

**Comment by connorjclark** [comment]
I agree, it seems the expectation is that "ar" alone is latin numerals. LH is doing things correctly here already.

**Comment by connorjclark** [comment]
Some of this code was recently modified via #13830

We should make sure we are using the number formatter for all numbers in the report. Some places we're (probably) missing today: the gauge, the score legend.
